controllers/usersignup/usersignup_controller_test.go

```diff
@@ -4463,6 +4463,76 @@ func TestUserSignupStatusNotReady(t *testing.T) {
 				Reason: "UserIsActive",
 			})
 	})
+
+	// If a space is updating, keep usersignups that have already completed
+	t.Run("keep usersignups while space is updating", func(t *testing.T) {
+		// given
+		space.Status.Conditions = []toolchainv1alpha1.Condition{{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: corev1.ConditionFalse,
+			Reason: toolchainv1alpha1.SpaceUpdatingReason,
+		}}
+		r, req, _ := prepareReconcile(t, userSignup.Name, NewGetMemberClusters(member), userSignup, mur, space, spacebinding, commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true)), baseNSTemplateTier, deactivate30Tier)
+		// when
+		res, err := r.Reconcile(context.TODO(), req)
+		require.NoError(t, err)
+		require.Equal(t, reconcile.Result{}, res)
+		// then
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, userSignup)
+		require.NoError(t, err)
+		test.AssertConditionsMatch(t, userSignup.Status.Conditions,
```

Comment:
I know that you copied the previous test cases, so the bug was already there, but we shouldn't use the same object for retrieving the actual UserSignup resource, because it influences the following subtests. We should use separate object instead.
```suggestion
		actualUserSignup := &toolchainv1alpha1.UserSignup{}
		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, actualUserSignup)
		require.NoError(t, err)
		test.AssertConditionsMatch(t, actualUserSignup.Status.Conditions,
```
I would be really thankful if you could update also the previous two test cases. When you update it, you will see that this one will start failing - it's just because the UserSignup won't have the needed condition upfront.

Sorry about the inconvenience. In most of the cases we use custom assertions with fluent-api, however, we haven't got to it for UserSignup yet.

Reply:
I did some cleanup and refactoring of these tests in 583f076 - should remove the dependency between tests.